packages/svelte-cloudinary/src/helpers/getCldImageUrl.ts

```diff
@@ -1,57 +1,31 @@
-import { constructCloudinaryUrl } from '@cloudinary-util/url-loader';
-import { checkCloudinaryCloudName } from '../cloudinary.js';
-import type { ImageOptions, ConfigOptions, AnalyticsOptions } from '@cloudinary-util/url-loader';
-
+import { mergeGlobalConfig } from '../config';
 import {
-	SVELTE_CLOUDINARY_ANALYTICS_ID,
-	SVELTE_CLOUDINARY_VERSION,
-	SVELTE_VERSION
-} from '../constants/analytics.js';
-/**
- * getCldImage
- */
-
-export interface GetCldImageUrlOptions extends ImageOptions {}
-export interface GetCldImageUrlConfig extends ConfigOptions {}
-export interface GetCldImageUrlAnalytics extends AnalyticsOptions {}
+	constructCloudinaryUrl,
+	type AnalyticsOptions,
+	type ConfigOptions,
+	type ImageOptions,
+} from '@cloudinary-util/url-loader';
 
-export interface GetCldImageUrl {
-	options: GetCldImageUrlOptions;
-	config?: GetCldImageUrlConfig;
-	analytics?: GetCldImageUrlAnalytics;
-}
-
-/**
- * Generates the Cloudinary url for the assets
- * based on the configuration passed to the function
- * @returns string
- */
 export function getCldImageUrl(
 	options: ImageOptions,
-	config?: ConfigOptions,
-	analytics?: AnalyticsOptions
+	configOverride?: ConfigOptions,
+	analyticsOverride?: AnalyticsOptions,
 ) {
-	// Validation
-	checkCloudinaryCloudName(import.meta.env.VITE_PUBLIC_CLOUDINARY_CLOUD_NAME);
+	const { config, analytics } = mergeGlobalConfig(
+		configOverride,
+		analyticsOverride,
+	);
 
 	return constructCloudinaryUrl({
-		options,
-		config: Object.assign(
-			{
-				cloud: {
-					cloudName: import.meta.env.VITE_PUBLIC_CLOUDINARY_CLOUD_NAME
-				}
-			},
-			config
-		),
-		analytics: Object.assign(
-			{
-				sdkCode: SVELTE_CLOUDINARY_ANALYTICS_ID,
-				sdkSemver: SVELTE_CLOUDINARY_VERSION,
-				techVersion: SVELTE_VERSION,
-				product: 'B'
+		analytics,
+		config,
+		options: {
+			...options,
+			crop: options.crop || {
+				type: 'fill',
+				gravity: 'center',
```

Comment:
we've been using `auto` as the default for the gravity. this allows the AI to kick in and automatically handle moving the frame based on the subject. are you intentionally using 'center' here?

Reply:
are the defaults applied in `constructCloudinaryUrl`? I just realised you aren't doing this in next-cloudinary so perhaps is left over from v1

Reply:
yeah but it's more complicated than just a simple default: https://github.com/cloudinary-community/cloudinary-util/blob/main/packages/url-loader/src/plugins/cropping.ts#L228

Reply:
so would it be safe to remove the entire default block I have there? (the fill, gravity etc)